drivers/gles2/shader_compiler_gles2.cpp

```diff
@@ -754,12 +754,21 @@ String ShaderCompilerGLES2::_dump_node_code(SL::Node *p_node, int p_level, Gener
 				} break;
 
 				case SL::OP_MOD: {
-
-					code += "mod(float(";
+					code += "(";
 					code += _dump_node_code(op_node->arguments[0], p_level, r_gen_code, p_actions, p_default_actions, p_assigning);
-					code += "), float(";
+					code += " < 0 ? ";
+					code += "-int(0.5 + mod(float(-";
+					code += _dump_node_code(op_node->arguments[0], p_level, r_gen_code, p_actions, p_default_actions, p_assigning);
```

Comment:
The ternary operator business might be more readable if you save the output of the `_dump_node_code` calls for both argument 0 and 1 in variables, and use those to construct `code`.

Reply:
Could you also squash the commits together eventually?

Reply:
I realized I was being silly building on the original implementation anyway so I replaced the whole thing. However, I can't figure out how to do the squashing; the only thing I've found is the "Squash and Merge" option for a PR but I don't have permission to merge here obviously.

Reply:
See our how-to on https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase